lib/catalog/registry/ProductsRegistry.js

```diff
@@ -0,0 +1,37 @@
+import Registry from './Registry.js';
+
+/**
+ * @template {Record<string, object>} Cache
+ * @extends {Registry<Cache & { products: Record<string, import('#types').Product> }>}
+ */
+export default class ProductsRegistry extends Registry {
+    /**
+     * @param {string} dataFolder
+     * @param {Array<keyof Cache>} cacheFiles
+     */
+    constructor(dataFolder, cacheFiles) {
+        super(dataFolder, ['products', ...cacheFiles]);
+    }
+
+    /**
+     * @param {string} id - product id
+     * @returns {import('#types').Product}
+     */
+    getOrCreateProduct (id) {
+        let instance = this.cache.products[id];
+
+        if (!instance) {
+            instance = {};
+        }
+
+        return instance;
+    }
+
+    /**
+     * @param {string} id
+     * @returns {boolean}
+     */
+    hasProduct (id) {
+        return id in this.products;
```

Comment:
This method is no longer used and has not actual logic because you now have this.cache.products